libsql/src/local/database.rs

```diff
@@ -476,21 +476,40 @@ impl Database {
     ) -> Result<u32> {
         let mut nr_retries = 0;
         loop {
-            let client = reqwest::Client::new();
-            let mut builder = client.post(uri.to_owned());
+            // TODO(lucio): add custom connector + tls support here
+            let client = hyper::client::Client::builder().build_http::<hyper::Body>();
+
+            let mut req = http::Request::post(uri.clone());
+
             match auth_token {
                 Some(ref auth_token) => {
-                    builder = builder
-                        .header("Authorization", format!("Bearer {}", auth_token.to_owned()));
+                    let auth_header =
+                        http::HeaderValue::try_from(format!("Bearer {}", auth_token.to_owned()))
+                            .unwrap();
+
+                    req.headers_mut()
+                        .expect("valid http request")
+                        .insert("Authorization", auth_header);
                 }
                 None => {}
             }
-            let res = builder.body(frame.to_vec()).send().await.unwrap();
+
+            // TODO(lucio): convert this to use bytes to make this clone cheap, it should be
+            // to possible use BytesMut when reading frames from the WAL and efficiently use Bytes
+            // from that.
+            let req = req.body(frame.clone().into()).expect("valid body");
```

Comment:
body wouldn't accept frame.to_vec()?

Reply:
I found `to_vec` to not be clear since, its already a `Vec<u8>` so this instead reads copy the frame and convert it into the type accepted by body (which in this case is hyper::Body). This should still compile to the same thing.